packages/camera/camera_android/lib/src/android_camera.dart

```diff
@@ -216,7 +216,11 @@ class AndroidCamera extends CameraPlatform {
 
   @override
   Future<XFile> stopVideoRecording(int cameraId) async {
-    final String path = await _hostApi.stopVideoRecording();
+    final String? path = await _hostApi.stopVideoRecording();
+    if (path == null) {
+      throw StateError(
```

Comment:
Is it possible for a plugin client to prevent this case from happening? If not, `StateError` isn't a useful thing to throw; per docs "An Error object represents a program failure that the programmer should have avoided." If so, the error should explain what the developer did incorrectly.

If this is unforeseeable, then it should throw some kind of Exception, not an Error, so that it can be caught and handled at runtime.